File: src/main/java/ua/softgroup/matrix/server/persistent/repository/TimeAuditRepository.java
package ua.softgroup.matrix.server.persistent.repository;

import org.springframework.data.repository.CrudRepository;
import ua.softgroup.matrix.server.persistent.entity.TimeAudit;
import ua.softgroup.matrix.server.persistent.entity.User;

import java.time.LocalDateTime;
import java.util.Set;

public interface TimeAuditRepository extends CrudRepository<TimeAudit, Long> {

    Set<TimeAudit> findByAdder(User adder);

    Set<TimeAudit> findByCreationDateBetween(LocalDateTime start, LocalDateTime end);
}

File: src/main/scala/ua/softgroup/matrix/server/persistent/repository/TimeAuditRepository.scala
package ua.softgroup.matrix.server.persistent.repository

import java.time.LocalDateTime
import java.util

import org.springframework.data.repository.CrudRepository
import ua.softgroup.matrix.server.persistent.entity.{TimeAudit, User}

/**
  * @author Oleksandr Tyshkovets <sg.olexander@gmail.com> 
  */
trait TimeAuditRepository extends CrudRepository[TimeAudit, java.lang.Long] {

  def findByPrincipalId(principalId: Long): util.Set[TimeAudit]

  def findByCreationDateBetween(start: LocalDateTime, end: LocalDateTime): util.Set[TimeAudit]

}
